rules/react.js

```diff
@@ -104,5 +104,108 @@ module.exports = {
     // expecially when data is client side derived.
     'react/no-array-index-key': 'warn',
+
+    // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-sort-props.md
+    // AirBnB
+    // 'react/jsx-sort-props': ['off', {
+    //   ignoreCase: true,
+    //   callbacksLast: false,
+    //   shorthandFirst: false,
+    //   shorthandLast: false,
+    //   noSortAlphabetically: false,
+    //   reservedFirst: true,
+    // }],
+    // ============
+    // Proposed:
+    'react/jsx-sort-props': [
+      'error',
+      {
+        // function will be last
+        // <Component isABC tel={5555555} onClick={this._handleClick} />
+        callbacksLast: true,
+        // ignore case when sorting
+        ignoreCase: true,
+        // Sort according to alphabetical order
+        noSortAlphabetically: false,
+        // shorthand like boolean prop will be first
+        // <Component a b c d="string" e="string" />
+        shorthandFirst: true,
+        shorthandLast: false,
+        // DOM's reserved props will be first
+        // <Component className="x" style={} isABC onClick={this._handleClick} />
+        reservedFirst: true,
+      },
+    ],
+
+    // ==========================
+    // ! IMPORTANT: CAN'T use --fix *
+    // ! Enable with precaution
+    // ==========================
+    // https://github.com/yannickcr/eslint-plugin-react/blob/843d71a432baf0f01f598d7cf1eea75ad6896e4b/docs/rules/jsx-sort-default-props.md
+    // AirBnB
+    // 'react/jsx-sort-default-props': ['off', {
+    //   ignoreCase: true,
+    // }],
+    // ============
+    // Proposed:
+    // In style with sorting component's props
+    // 'react/jsx-sort-default-props': [
+    //   'error',
+    //   {
+    //     ignoreCase: true,
+    //   },
+    // ],
+
+    // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-direct-mutation-state.md
+    // AirBnB
+    // 'react/no-direct-mutation-state': 'off',
+    // ============
+    // Proposed:
+    // Enforce immutability
+    'react/no-direct-mutation-state': 'error',
+
+    // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/forbid-component-props.md
+    // AirBnB
+    // 'react/forbid-component-props': ['off', { forbid: [] }],
+    // ============
+    // Default value: className & style
+    // Proposed:
+    // Enforce styling inside styled-component rather than class
+    // We can pass custom props to styled-component and assign logic there
+    // Span allowed for our icon set like linear icons but we can also specify the required icon
+    // inside styled-component attrs but that means we need to create a lot of icon component.
+    // If no additional styling required, using className for span icon probably good enough.
+    // Also forbid using "style" as component's props.
+    'react/forbid-component-props': [
+      'error',
+      {
+        forbid: [{ propName: 'className', allowedFor: ['span'] }, { propName: 'style' }],
```

Comment:
Linear Icon could attach by props by styled-component
```
const LargeCircle = styled.span.attrs({ className: 'lnr lnr-circle' })'
  font-size: 36px;
  color: #ff0000;
`
```
So I think className should be very generic already with JSX and should not strict on it.